test/indexing.jl

```diff
@@ -71,64 +59,55 @@ using Test, DataFrames
     @test df[Not(Not(1:2)), Not(1:0)] == DataFrame(a=1:2, b=4:5, c=7:8)
 
     @test df[:, 1] == [1, 2, 3]
-    @test df[:, 1] !== df[1]
+    @test df[:, 1] !== df[!, 1]
     @test df[:, 1:2] == DataFrame(a=1:3, b=4:6)
     @test df[:, r"[ab]"] == DataFrame(a=1:3, b=4:6)
     @test df[:, Not(r"c")] == DataFrame(a=1:3, b=4:6)
-    @test eachcol(df[:, 1:2])[1] !== df[1]
+    @test eachcol(df[:, 1:2])[1] !== df[!, 1]
     @test df[:, :] == df
     @test df[:, r""] == df
     @test df[:, Not(Not(r""))] == df
-    @test eachcol(df[:, :])[1] !== df[1]
-    @test eachcol(df[:, r""])[1] !== df[1]
-    @test eachcol(df[:, Not([])])[1] !== df[1]
+    @test eachcol(df[:, :])[1] !== df[!, 1]
+    @test eachcol(df[:, r""])[1] !== df[!, 1]
+    @test eachcol(df[:, Not([])])[1] !== df[!, 1]
 
     @test df[Not(Int[]), 1] == [1, 2, 3]
-    @test df[Not(Int[]), 1] !== df[1]
+    @test df[Not(Int[]), 1] !== df[!, 1]
     @test df[Not(Int[]), 1:2] == DataFrame(a=1:3, b=4:6)
     @test df[Not(Int[]), r"[ab]"] == DataFrame(a=1:3, b=4:6)
     @test df[Not(Int[]), Not(r"c")] == DataFrame(a=1:3, b=4:6)
-    @test eachcol(df[Not(Int[]), 1:2])[1] !== df[1]
+    @test eachcol(df[Not(Int[]), 1:2])[1] !== df[!, 1]
     @test df[Not(Int[]), :] == df
     @test df[Not(Int[]), r""] == df
     @test df[Not(Int[]), Not(Not(r""))] == df
-    @test eachcol(df[Not(Int[]), :])[1] !== df[1]
-    @test eachcol(df[Not(Int[]), r""])[1] !== df[1]
-    @test eachcol(df[Not(Int[]), Not([])])[1] !== df[1]
+    @test eachcol(df[Not(Int[]), :])[1] !== df[!, 1]
+    @test eachcol(df[Not(Int[]), r""])[1] !== df[!, 1]
+    @test eachcol(df[Not(Int[]), Not([])])[1] !== df[!, 1]
 end
 
-@testset "getindex df[col] and df[cols]" begin
+@testset "getindex df[!, col]" begin
     x = [1, 2, 3]
     df = DataFrame(x=x, copycols=false)
     @test df.x === x
-    @test df[:x] === x
-    @test df[[:x]].x !== x
```

Comment:
Restore equivalent of these tests if you add `df[, cols]`. Same for `view` below.

Reply:
done